source/common/http/filter/cors_filter.cc

```diff
@@ -0,0 +1,140 @@
+#include "common/http/filter/cors_filter.h"
+
+#include "envoy/http/codes.h"
+
+#include "common/common/enum_to_int.h"
+#include "common/http/header_map_impl.h"
+#include "common/http/headers.h"
+
+namespace Envoy {
+namespace Http {
+
+CorsFilter::CorsFilter(CorsFilterConfigConstSharedPtr config)
+    : config_(config), is_cors_request_(false) {}
+
+CorsFilter::~CorsFilter() {}
+
+FilterHeadersStatus CorsFilter::decodeHeaders(HeaderMap& headers, bool) {
+  initialize();
```

Comment:
would it maks sense to make this initializeCorsPolicy and only call if the origin header is present on an OPTIONS request?

Reply:
That is definitely better!